tiledb/sm/array_schema/array_schema.cc

```diff
@@ -946,8 +951,13 @@ void ArraySchema::set_name(const std::string& name) {
   name_ = name;
 }
 
-void ArraySchema::set_capacity(uint64_t capacity) {
+Status ArraySchema::set_capacity(uint64_t capacity) {
+  if (array_type_ == ArrayType::SPARSE && capacity == 0)
+    return LOG_STATUS(Status_ArraySchemaError(
```

Comment:
Please change to `throw` and keep the `void` signature. The C API [`api_entry*` mechanism](https://github.com/TileDB-Inc/TileDB/blob/9e4883263b753e8b3b2bd600222119a0f17c48c3/tiledb/sm/c_api/api_exception_safety.h) will catch and translate the error.

Reply:
(FYI - if that's not possible for some existing function, use `throw_if_not_ok` at the boundary)

Reply:
Don't use `Status` in new code. Remove it from old code when modifying it.